examples/format_osmfish.py

```diff
@@ -0,0 +1,202 @@
+def cached_read_fn(file_path) -> np.ndarray:
+    return np.load(file_path)
+
+
+class osmFISHTile(FetchedTile):
+
+    def __init__(
+            self,
+            file_path: str,
+            coordinates: Mapping[Union[str, Coordinates], Union[Number, Tuple[Number, Number]]],
+            z: int
+    ) -> None:
+        """Initialize a TileFetcher for osmFISH Data"""
+        self.file_path = file_path
+        self.z = z
+        self._coordinates = coordinates
+
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        return self.tile_data().shape
+
+    @property
+    def coordinates(self) -> Mapping[Union[str, Coordinates], Union[Number, Tuple[Number, Number]]]:
+        """For this single field of view test experiment, we expose dummy coordinates"""
+        return self._coordinates
+
+    @property
+    def format(self) -> ImageFormat:
+        return ImageFormat.NUMPY
+
+    def tile_data(self) -> np.ndarray:
+        return cached_read_fn(self.file_path)[self.z]  # slice out the correct z-plane
+
+
+class osmFISHTileFetcher(TileFetcher):
+
+    @starfish.util.try_import.try_import({"yaml"})
+    def __init__(self, input_dir: str, metadata_yaml) -> None:
+        """Implement a TileFetcher for an osmFISH experiment.
+
+        This TileFetcher constructs spaceTx format for one or more fields of view, where
+        `input_dir` is a directory containing all .npy image files and whose file names have the
+        following structure:
+
+        Hybridization<round>_<target>_fov_<fov_number>.npy
+
+        Notes
+        -----
+        - osmFISH is a non-multiplex method. As such, each target is specified by a
+          (channel, round) tuple. The files do not contain channel information,
+        - The spatial organization of the fields of view are not known to the starfish developers,
+          so they are filled by dummy coordinates
+        - This TileFetcher is specifically tailored to the gene panel used for a specific
+          experiment. Generalization of this TileFetcher will require reimplementation of the
+          `channel_map` method.
+        """
+        import yaml
+
+        with open(metadata_yaml, 'r') as f:
+            self.osmfish_metadata = yaml.load(f)
+        self.num_z = self.osmfish_metadata['ImageProperties']['HybImageSize']['zcount']
+        self.input_dir = input_dir
+
+    @property
+    def channel_map(self) -> Mapping[str, int]:
+        return {
+            "Quasar570": 0,
+            "CF610": 1,
+            "Quasar670": 2,
+        }
+
+    @property
+    def target_map(self) -> Mapping:
+        """
+        Parse the metadata to map channel number and round to the correct target, which identifies
+        the right tile
+        """
+        parsed_metadata = {}
+        for round_, round_data in self.osmfish_metadata['HybridizationsInfos'].items():
+            round_match = re.match("Hybridization(\d{1,2})", round_)
+            if round_match is not None:
```

Comment:
Do we ever expect round_match to be `None`?  If not, then we should just fatal if it's `None`.

Reply:
Yea, there are some values in the metadata that we don't have the files for, so some of these will fail. I made a note to explain this better.

Reply:
Got it, understood.

I generally prefer (for loops):

```
if round_match is None:
    continue
```

Rationale:
1. reduces levels of indents
2. we know we don't need to follow the control flow past the if-statement for this case.